### PR TITLE
Pixify repo for better dev and package management

### PR DESCRIPTION
**Pull request by KedoKudo**
This PR introduces the following changes:

- Switch to pure `pyproject.toml` to conform with modern Python packaging
- Enable standard repo management
- Add missing module/class/function docstrings
- Add missing type-hinting
- Fixing various bugs related to edge cases
- Add `pytest` base structures and placeholders
- Add Github actions for unit test and package building and publishing, configured corresponding services
> EWM item: [10366](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20%28Change%20Management%29#action=com.ibm.team.workitem.viewWorkItem&id=10366)

## Testing instructions

- Check out the feature branch
- Make sure the environment is clean and no conda environment is activated
- Make sure `pixi` is properly installed and updated
- Run `pixi install` to setup the development environment
- Run `pixi run hyperctui` to start the GUI application
- Run `pixi run test` to start the dummy test suite

---
## Summary [AI]

This pull request includes several significant changes to the repository, focusing on automation, configuration, and documentation updates. The most important changes include the addition of GitHub Actions workflows, configuration files for dependency management, and updates to the project's documentation and licensing.

### Automation and CI/CD:

* Added a new GitHub Actions workflow for packaging and deployment, including steps for building and uploading conda and PyPI packages (`.github/workflows/package.yml`).
* Introduced a unit test workflow to run tests and upload coverage reports to Codecov (`.github/workflows/unittest.yml`).
* Created a workflow to update lockfiles on a schedule and create pull requests with the updates (`.github/workflows/update-lockfiles.yml`).

### Dependency Management:

* Configured Dependabot to automate dependency updates for GitHub Actions (`.github/dependabot.yml`).
* Added pre-commit hooks to enforce coding standards and prevent common issues (`.pre-commit-config.yaml`).

### Documentation and Licensing:

* Updated the `README.md` with detailed installation instructions, development guidelines, and contribution guidelines.
* Added the MIT License to the project (`LICENSE`).

These changes enhance the project's automation, ensure consistent code quality, and provide clear guidelines for contributors.

### CODE REVIEW

**Review by JeanBilheux — APPROVED** [review]
Everything works as expected.

File: conda.recipe/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    - pandas
+    - pillow
+    - scikit-image
+    - loguru
+
+
+about:
+  home: {{ url }}
+  license: {{ license }}
+  license_family: GPL
```
[the remark this answers is not in the document]
Comment (JeanBilheux) [reply conda.recipe/meta.yaml]: That's perfectly fine.
